File: app/src/main/java/net/lyxnx/carcheck/CalculatorsActivity.java
package net.lyxnx.carcheck;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import net.lyxnx.carcheck.adapter.CalculatorsPagerAdapter;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager2.widget.ViewPager2;

public class CalculatorsActivity extends AppCompatActivity {

    private ViewPager2 pager;
    private MenuItem previousItem;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calcs);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        BottomNavigationView navigation = findViewById(R.id.navigation);

        navigation.setOnNavigationItemSelectedListener(item -> {
            switch (item.getItemId()) {
                case R.id.action_journey:
                    pager.setCurrentItem(CalculatorsPagerAdapter.JOURNEY);
                    return true;
                case R.id.action_mpg:
                    pager.setCurrentItem(CalculatorsPagerAdapter.MPG);
                    return true;
                case R.id.action_finance:
                    pager.setCurrentItem(CalculatorsPagerAdapter.FINANCE);
                    return true;
                default:
                    return false;
            }
        });

        CalculatorsPagerAdapter adapter = new CalculatorsPagerAdapter(this);

        pager = findViewById(R.id.main_content);
        pager.setOffscreenPageLimit(1);
        pager.setAdapter(adapter);
        pager.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
            @Override
            public void onPageSelected(int position) {
                if (previousItem != null) {
                    previousItem.setChecked(false);
                } else {
                    navigation.getMenu().getItem(0).setChecked(false);
                }

                MenuItem item = navigation.getMenu().getItem(position);
                item.setChecked(true);
                previousItem = item;
                setTitle(getString(R.string.title_calculators, adapter.getTitle(position)));
            }
        });
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            this.finish();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/net/lyxnx/carcheck/CalculatorsActivity.kt
package net.lyxnx.carcheck

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.viewpager2.widget.ViewPager2
import androidx.viewpager2.widget.ViewPager2.OnPageChangeCallback
import com.google.android.material.bottomnavigation.BottomNavigationView
import net.lyxnx.carcheck.adapter.CalculatorsPagerAdapter

class CalculatorsActivity : AppCompatActivity() {

    private lateinit var pager: ViewPager2
    private var previousItem: MenuItem? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calcs)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)

        pager = findViewById(R.id.main_content)

        val navigation = findViewById<BottomNavigationView>(R.id.navigation)
        navigation.setOnNavigationItemSelectedListener { item: MenuItem ->
            when (item.itemId) {
                R.id.action_journey -> {
                    pager.currentItem = CalculatorsPagerAdapter.JOURNEY
                    return@setOnNavigationItemSelectedListener true
                }
                R.id.action_mpg -> {
                    pager.currentItem = CalculatorsPagerAdapter.MPG
                    return@setOnNavigationItemSelectedListener true
                }
                R.id.action_finance -> {
                    pager.currentItem = CalculatorsPagerAdapter.FINANCE
                    return@setOnNavigationItemSelectedListener true
                }
                else -> return@setOnNavigationItemSelectedListener false
            }
        }

        val adapter = CalculatorsPagerAdapter(this)

        pager.offscreenPageLimit = 1
        pager.adapter = adapter
        pager.registerOnPageChangeCallback(object : OnPageChangeCallback() {
            override fun onPageSelected(position: Int) {
                if (previousItem != null) {
                    previousItem!!.isChecked = false
                } else {
                    navigation.menu.getItem(0).isChecked = false
                }

                val item = navigation.menu.getItem(position)
                item.isChecked = true
                previousItem = item
                title = getString(R.string.title_calculators, adapter.getTitle(position))
            }
        })
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if (item.itemId == android.R.id.home) {
            finish()
            true
        } else super.onOptionsItemSelected(item)
    }

}